### ISSUE TITLE
Unauthenticated - AuthenticationException - No matching records

### ISSUE THREAD

**Issue by PengKuang** [grokability/snipe-it#8697]
#### Please confirm you have done the following before posting your bug report:

- [ x] I have enabled debug mode 
- [ x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

**Describe the bug**
Cannot log in successfully using existing credentials to the system

PDOException: SQLSTATE[42522] Column not found: 1054 Unknow column 'ad_append_domain' in 'field list'

**To Reproduce**
1. Go to 'http://my custom domain/login'
2. Click on 'username' and 'password' to input the credentials and then click on the 'login' button
3. Scroll down to debug console
4. See 3 errors as in the screenshots attached

**Expected behavior**
logged in successfully 

**Screenshots**
<img width="529" alt="snipe-it error 0" src="https://user-images.githubusercontent.com/31425596/98504583-5c0f2c80-2292-11eb-81b8-0cd7c4ef1e88.PNG">

**Server (please complete the following information):**
 - Snipe-IT Version: 5.0.5 pre
 - OS: CentOS 8
 - Web Server: Apache 2.4.37
 - PHP Version: 7.2.34 
- database server: Mysql

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: chrome & firefox
 - Version: unknown

**Error Messages**
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error
- If a stacktrace is provided in the error, include that too.
- Any errors that appear in your browser's error console.
- Confirm whether the error is reproducible on the demo: https://snipeitapp.com/demo.
- Include any additional information you can find in `storage/logs` and your webserver's logs.

**Additional context**
- Is this a fresh install or upgrade?
This is an fresh installation on a new server with separate database server since the old server OS is too old and the software cannot be upgraded directly there; and the backup of old database has been imported to the new database
- What OS and web server you're running Snipe-IT on
centos 8 and apache 2
- What method you used to install Snipe-IT  
manual installation
- Include what you've done so far in the installation, and if you got any error messages along the way.
followed the official guide, done installation, configuration, dependency and server set up
when generate the key and type the server address in the browser, encountered this error
- Indicate whether or not you've manually edited any data directly in the database
none

<img width="524" alt="snipe-it error 01" src="https://user-images.githubusercontent.com/31425596/98500426-6e379d80-2287-11eb-86a7-ade186356d75.PNG">
<img width="529" alt="snipe-it error 02" src="https://user-images.githubusercontent.com/31425596/98500432-7099f780-2287-11eb-88bf-6cf7f748152d.PNG">
<img width="515" alt="snipe-it error 03" src="https://user-images.githubusercontent.com/31425596/98500434-72fc5180-2287-11eb-84b4-abd44a412c0c.PNG">

**Comment by PengKuang** [comment]
btw, I didn't try 'php artisan migration' since I'm using a new installation to connect to a database with the old data.

I backed up the previous database and imported it to the new database. 

So I'm not sure if I run the migration command, it will erase the tables and records I already imported to the new database.

Also,  since I couldn't upgrade the old snipeit due to retiring server OS centos 6, I installed this new one on a new server with centos 8. Should I replace the new app key using the old app key in this scenario?

**Comment by PengKuang** [comment]
I eventually ran the php artisan migrate command and got logged in successfully. However, I encountered a new error。

It's super weird that there are correct records when I logged in as a regular user but when I logged in as an admin instead, there are no records found. I can see a total number but all specific records are empty. It says no matching records.

what I've done is:

copied the files in the storage/private_uploads from old server to the new server
copied the files in the public/uploads from old server to the new server
copied the oauth public and private keys from old server and replace the ones in the new server

![image](https://user-images.githubusercontent.com/31425596/98522172-56740f80-22af-11eb-8c45-8cdd8c8a23c1.png)

![image](https://user-images.githubusercontent.com/31425596/98518221-0e062300-22aa-11eb-9c61-bcec54fe5b67.png)

when I logged in as a regular user, I can see my records correctly although still having the same error:

![image](https://user-images.githubusercontent.com/31425596/98522687-f6ca3400-22af-11eb-9601-e10a4bcd5ede.png)

I'm not sure if this is because the old server is https and the new server I haven't installed a certificate there; it is http

**Comment by bhillebrand-enee** [comment]
**Same here:**
We are getting HTTP 401 on XHR datatable requests following an upgrade from Snipe 4.6.x to 5.0.4.
Apache 2.4, PHP 7.3.23 with a MariaDB 10.3.25 db backend.

**Have done the usual troubleshooting steps:** 
- composer dump-autoload
- composer clear-compiled
- php artisan config:clear
- php artisan view:clear
- php artisan cache:clear
- php artisan route:clear

[
![401 on XHR](https://user-images.githubusercontent.com/31837021/98528616-78c85600-22cc-11eb-9000-2a05b63081f5.png)
](url)

**Comment by snipe** [comment]
This is going to be a bit of a cut and paste job, so forgive me if I reference the this ticket in the list. Sorry, just still trying to get as much info as I can. This was first (I believe) reported in #8458.

Thanks for reporting this, folks. As has been discussed in other tickets (specifically #8458 but probably also #8644, #8587, and #8650), this seems like it *must* be an environmental issue, as we've been unable to reproduce this on *any* of our hosted customers. 

Some folks have found that running `php upgrade.php` worked for them, which means something didn't get cleared or run that the `upgrade.php` script clears or runs for you. https://github.com/snipe/snipe-it/issues/8458#issuecomment-723416695

Other folks have found that they were missing some of the required extensions: https://github.com/snipe/snipe-it/issues/8458#issuecomment-723856070
https://snipe-it.readme.io/docs/requirements#general-system-requirements

<blockquote><img src="/favicon.ico" width="48" align="right"><div>Snipe-IT Documentation</div><div><strong><a href="https://snipe-it.readme.io/docs/requirements#general-system-requirements">Requirements</a></strong></div><div>The requirements for Snipe-IT are fairly standard for a modern web server, and most servers will have most or all of these components already installed. If for some reason you don't have them installed, they are relatively easy to add.      </div></blockquote>

**Comment by snipe** [comment]
Please also see my summary here: https://github.com/snipe/snipe-it/issues/8458#issuecomment-724443005

**Comment by bhillebrand-enee** [comment]
Hi @snipe!

Thanks for the speedy response. Following your advice  I decided to double check that everything was correct as I was in the course of compiling information that would help diagnose the exact environmental issue. 

TLDR: During the course of this I did some troubleshooting which included completely removing the vendor packages folder to ensure that Composer populated a fresh copy of the files. Upon re-running php upgrade.php Snipe started working normally. 

**Would recommend that you guys include (in upgrade.php) an rm -rf statement for the vendor packages folder, as this would eliminate the risk of vendor package corruption. Also, it may be a good idea to specifically require Composer 2.x before running; I have a hunch that my initially running the update through Composer 1.x may have contributed to the inconsistent vendor packages state.**
